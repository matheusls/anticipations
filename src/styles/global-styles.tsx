import { createGlobalStyle } from 'styled-components';
import normalize from 'styled-normalize';

const GlobalStyles = createGlobalStyle`
  ${normalize}

  * {
    box-sizing: border-box;
  }

  html, body {
    font-size: 1em;
  }

  body {
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  }
`;

export default GlobalStyles;
