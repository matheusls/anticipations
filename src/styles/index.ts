export { default as GlobalStyles } from './global-styles';
export { default as Theme } from './theme';
