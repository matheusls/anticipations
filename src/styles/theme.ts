import { DefaultTheme } from 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      black: string;
      blue: string;
      blueDark: string;
      blueLight: string;
      blueLighter: string;
      gray: string;
      grayDark: string;
      grayLight: string;
      white: string;
    };
    fontSizes: {
      xsmall: string;
      small: string;
      medium: string;
      large: string;
    };
  }
}

const Theme: DefaultTheme = {
  colors: {
    black: '#000',
    blue: '#5d9bec',
    blueDark: '#3d75bb',
    blueLight: '#66afe9',
    blueLighter: '#dde6e9',
    gray: '#cecece',
    grayDark: '#656565',
    grayLight: '#f2f2f2',
    white: '#fff',
  },
  fontSizes: {
    xsmall: '0.688rem',
    small: '0.875rem',
    medium: '1rem',
    large: '1.5rem',
  },
};

export default Theme;
