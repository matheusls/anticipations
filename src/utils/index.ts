export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
