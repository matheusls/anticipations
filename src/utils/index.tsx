import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
