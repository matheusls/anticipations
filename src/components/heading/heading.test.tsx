import { render, screen } from 'tests/utils';

import Heading, { HeadingLevels } from './heading';

import { Theme } from 'styles';

const { colors, fontSizes } = Theme;

const defaultStyles = {
  color: colors.grayDark,
  fontSize: fontSizes.xlarge,
  fontStyle: 'normal',
};

const headingLevels: HeadingLevels[][] = [[1], [2], [3], [4], [5], [6]];

describe('<Heading />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Heading level={1}>Heading Component</Heading>,
    );

    expect(container).toMatchSnapshot();
  });

  it.each(headingLevels)(
    '%# should render Heading level %p correctly',
    (level) => {
      render(<Heading level={level}>Heading Component</Heading>);

      expect(screen.getByRole('heading', { level })).toBeInTheDocument();
    },
  );

  it('should render Heading with correct text', () => {
    render(<Heading level={1}>Heading Component</Heading>);

    expect(
      screen.getByRole('heading', { name: /heading component/i }),
    ).toBeInTheDocument();
  });

  it('should render Heading with default styles', () => {
    render(<Heading level={1}>Heading Component</Heading>);

    const heading = screen.getByRole('heading', { name: /heading component/i });

    const { color, fontSize, fontStyle } = defaultStyles;

    expect(heading).toHaveStyleRule('color', color);
    expect(heading).toHaveStyleRule('font-size', fontSize);
    expect(heading).toHaveStyleRule('font-style', fontStyle);
  });
});
