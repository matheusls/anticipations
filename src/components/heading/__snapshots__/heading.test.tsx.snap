// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
.c0 {
  color: #656565;
  display: block;
  font-size: 1.5rem;
  font-style: normal;
  font-weight: 700;
  line-height: 1.5;
  margin-bottom: 22px;
  text-transform: none;
  width: 100%;
}

<div>
  <h1
    class="c0"
  >
    Heading Component
  </h1>
</div>
`;
